direct.go

```diff
@@ -51,23 +50,29 @@ type direct struct {
 	maxCacheSize        int
 	cacheSaveInterval   time.Duration
 	toCache             chan masquerade
+	defaultProviderID   string
+	providers           map[string]*Provider
+	ready               chan struct{}
+	readyOnce           sync.Once
 }
 
 // Configure sets the masquerades to use, the trusted root CAs, and the
 // cache file for caching masquerades to set up direct domain fronting.
-func Configure(pool *x509.CertPool, masquerades map[string][]*Masquerade, cacheFile string) {
+// defaultProviderID is used when a masquerade without a provider is
+// encountered (eg in a cache file)
+func Configure(pool *x509.CertPool, providers map[string]*Provider, defaultProviderID string, cacheFile string) {
```

Comment:
I wonder if there would ever be a need to have separate sets of trusted CAs by provider ... I think it's fine as-is, but since we've already got the patient open I figured I'd raise it.

Reply:
My gut response is no... yagni / if we wouldn't trust them to sign certain things, why trust them for others? But I think the case could be made that if they are separate, there could be more disjoint attack surfaces...

Reply:
Yeah, I'd leave as-is.